### ISSUE TITLE
libeccodes_f90.dylib load error: libquadmath.0.dylib could not be found (macOS Apple Silicon)

### ISSUE THREAD

**Issue by truedichotomy** [JuliaLang/julia#44506]
I tried to install `GRIB.jl` and encountered the following issue during precompile:

```
julia> Pkg.precompile()
Precompiling project...
  ✗ GRIB
  0 dependencies successfully precompiled in 2 seconds. 524 already precompiled.

ERROR: The following 1 direct dependency failed to precompile:

GRIB [b16dfd50-4035-11e9-28d4-9dfe17e6779b]

Failed to precompile GRIB [b16dfd50-4035-11e9-28d4-9dfe17e6779b] to /Users/gong/.julia/compiled/v1.9/GRIB/jl_BaDQDr.
ERROR: LoadError: InitError: could not load library "/Users/gong/.julia/artifacts/9ebc5e497534b4f1fb3aed45fce27c53bdb8938f/lib/libeccodes_f90.dylib"
dlopen(/Users/gong/.julia/artifacts/9ebc5e497534b4f1fb3aed45fce27c53bdb8938f/lib/libeccodes_f90.dylib, 0x0001): Library not loaded: @rpath/libquadmath.0.dylib
  Referenced from: /Users/gong/.julia/artifacts/9ebc5e497534b4f1fb3aed45fce27c53bdb8938f/lib/libeccodes_f90.dylib
  Reason: tried: '/Users/gong/.julia/artifacts/9ebc5e497534b4f1fb3aed45fce27c53bdb8938f/lib/../lib/libquadmath.0.dylib' (no such file), '/Users/gong/.julia/artifacts/9ebc5e497534b4f1fb3aed45fce27c53bdb8938f/lib/../lib/libquadmath.0.dylib' (no such file), '/Users/gong/GitHub/julia/usr/lib/libquadmath.0.dylib' (no such file), '/Users/gong/GitHub/julia/usr/bin/../lib/libquadmath.0.dylib' (no such file), '/usr/local/lib/libquadmath.0.dylib' (no such file), '/usr/lib/libquadmath.0.dylib' (no such file)
Stacktrace:
  [1] dlopen(s::String, flags::UInt32; throw_error::Bool)
    @ Base.Libc.Libdl ./libdl.jl:117
  [2] dlopen(s::String, flags::UInt32)
    @ Base.Libc.Libdl ./libdl.jl:117
  [3] macro expansion
    @ ~/.julia/packages/JLLWrappers/QpMQW/src/products/library_generators.jl:54 [inlined]
  [4] __init__()
    @ eccodes_jll ~/.julia/packages/eccodes_jll/nIoAN/src/wrappers/aarch64-apple-darwin-libgfortran5.jl:18
  [5] _include_from_serialized(path::String, depmods::Vector{Any})
    @ Base ./loading.jl:811
  [6] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String, depth::Int64)
    @ Base ./loading.jl:919
  [7] _require_search_from_serialized
    @ ./loading.jl:892 [inlined]
  [8] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1194
  [9] _require_prelocked(uuidkey::Base.PkgId)
    @ Base ./loading.jl:1087
 [10] macro expansion
    @ ./loading.jl:1067 [inlined]
 [11] macro expansion
    @ ./lock.jl:223 [inlined]
 [12] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1031
 [13] include
    @ ./Base.jl:426 [inlined]
 [14] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1418
 [15] top-level scope
    @ stdin:1
during initialization of module eccodes_jll
in expression starting at /Users/gong/.julia/packages/GRIB/6rlik/src/GRIB.jl:1
in expression starting at stdin:1
Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types ~/GitHub/julia/usr/share/julia/stdlib/v1.9/Pkg/src/Types.jl:67
 [2] precompile(ctx::Pkg.Types.Context, pkgs::Vector{String}; internal_call::Bool, strict::Bool, warn_loaded::Bool, already_instantiated::Bool, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API ~/GitHub/julia/usr/share/julia/stdlib/v1.9/Pkg/src/API.jl:1427
 [3] precompile
   @ ~/GitHub/julia/usr/share/julia/stdlib/v1.9/Pkg/src/API.jl:1060 [inlined]
 [4] #precompile#225
   @ ~/GitHub/julia/usr/share/julia/stdlib/v1.9/Pkg/src/API.jl:1057 [inlined]
 [5] precompile (repeats 2 times)
   @ ~/GitHub/julia/usr/share/julia/stdlib/v1.9/Pkg/src/API.jl:1057 [inlined]
 [6] top-level scope
   @ REPL[4]:1
```

```
julia> versioninfo()
Julia Version 1.9.0-DEV.150
Commit 811e534ca7 (2022-03-07 04:40 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.3.0)
  CPU: 10 × Apple M1 Pro
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores
```

**Comment by giordano** [comment]
Will be fixed by #44487

**Comment by fxcoudert** [comment]
In earlier versions of the GCC compiler for Apple Silicon, there was not support for `binary128` floating-point types, and therefore not `libquadmath`. In more recent versions, it is supported and the library is present.

It looks like you're using a `libeccodes_f90.dylib` that links against it, maybe because it was produced by a more recent compiler than you're using.
